R/rdeephaven/R/table_ops.R

```diff
@@ -0,0 +1,659 @@
+setGeneric(
+  "merge",
+  function(x, y = NULL, ...) {
+    standardGeneric("merge")
+  }
+)
+
+# We need many implementations of 'merge' because of the kind of args it should support
+
+base_merge <- function(x, y, ...) {
+  arg_list <- c(unlist(x), unlist(y), unlist(c(...)))
+  if (length(arg_list) == 1) {
+    return(arg_list[[1]])
+  }
+  unwrapped_arg_list <- lapply(arg_list, strip_s4_wrapping)
+  return(new("TableHandle", .internal_rcpp_object = unwrapped_arg_list[[1]]$merge(unwrapped_arg_list[2:length(unwrapped_arg_list)])))
+}
+
+#' @export
+setMethod(
+  "merge",
+  signature = c(x = "list", y = "missing"),
+  function(x, y = NULL) {
+    verify_type("x", x, "TableHandle", "Deephaven TableHandle", FALSE)
+    return(base_merge(x, y))
+  }
+)
+
+#' @export
+setMethod(
+  "merge",
+  signature = c(x = "TableHandle", y = "missing"),
+  function(x, y = NULL) {
+    return(base_merge(x, y))
+  }
+)
+
+#' @export
+setMethod(
```

Comment:
I do not think these merge methods are doing what is desired.  There is a big contortion to make `merge(t1,t2,t3)` work, but it never works right.  For example `merge(NULL, NULL, t1)` is within scope of what we want, but it will not work.  It isn't obvious how to even make it work.  On the other hand it is easy to make `merge(c(NULL,NULL,t1))` work.  If we don't have a compelling reason to support `merge(t1,t2,t3)` we should remove it and get a clean implementation that works as desired.

Reply:
I can make `merge(NULL, NULL, t1)` work, but I don't know why we should. I think that `merge(t1, t2, t3)`, `merge(c(t1, t2, t3))` and `merge(t1, c(t2, t3))` should work, the first and last cases coming from piping use cases. I can support NULL if needed. The class unions you sent may be the way to go. This is terrible.